mitosheet/src/components/LoadingIndicator.tsx

```diff
@@ -1,43 +1,275 @@
+            return 'Sorting a dataframe'
+        }
+        case StepType.Pivot: {
+            return 'Pivoting a dataframe'
+        }
+        case StepType.Merge: {
+            return 'Merging dataframes'
+        }
+        case StepType.Concat: {
+            return 'Concatenating dataframes'
+        }
+        case StepType.DropDuplicates: {
+            return 'Dropping duplicates'
+        }
+        case StepType.ChangeColumnDtype: {
+            return 'Changing a dtype'
+        }
+        case StepType.ChangeColumnFormat: {
+            return 'Formatting'
+        }
+        case StepType.SetCellValue: {
+            return 'Setting a cell value'
+        }
+        case StepType.BulkOldRename: {
+            return 'Bulk old rename'
+        }
+        case StepType.ExcelImport: {
+            return 'Import an Excel file'
+        }
+        case StepType.Graph: {
+            return 'Creating graph'
+        }
+        case StepType.GraphDuplicate: {
+            return 'Duplicating graph'
+        }
+        case StepType.GraphDelete: {
+            return 'Deleting graph'
+        }
+        case StepType.GraphRename: {
+            return 'Renaming graph'
+        }
+        case StepType.DeleteRow: {
+            return 'Deleting row'
+        }
+        case StepType.PromoteRowToHeader: {
+            return 'Promoting row to header'
+        }
+        case StepType.SplitTextToColumns: {
+            return 'Spliting column'
+        }
+        case StepType.Transpose: {
+            return 'Transposing dataframe'
+        }
+        case StepType.Melt: {
+            return 'Melting dataframe'
+        }
+        case UpdateType.Undo: {
+            return 'Undo previous edit'
+        }
+        case UpdateType.Redo: {
+            return 'Redo previous edit'
+        }
+        case UpdateType.Clear: {
+            return 'Clear all edits'
+        }
+        case UpdateType.ArgsUpdate: {
+            return 'Updating user profile';
+        }
+        case UpdateType.SaveAnalysisUpdate: {
+            return 'Saving current analysis'
+        }
+        case UpdateType.ReplayAnalysisUpdate: {
+            return 'Replaying previous analysis'
+        }
+        case UpdateType.CheckoutStepByIdxUpdate: {
+            return 'Checking out previous step'
+        }
+        case UpdateType.AppendUserFieldUpdate: {
+            return 'Updating user profile'
+        }
+        case UpdateType.SetUserFieldUpdate: {
+            return 'Updating user profile'
+        }
+        case UpdateType.UpdateFeedbackv2ObjObject: {
+            return 'Updating user profile'
+        }
+        case UpdateType.GoPro: {
+            return 'Activating Mito Pro'
+        }
+        case UpdateType.RenderCountUpdate: {
+            return 'Updating user profile'
+        }
+    }
+}
+
+const getMessageType = (messageType: string): StepType | UpdateType | undefined => {
+    if (isStepEvent(messageType)) {
```

Comment:
Do we want to keep this as `edit` event instead of introducing a new notion of `step` event?